SimTracker/TrackerHitAssociation/src/TrackerHitAssociator.cc

```diff
@@ -546,6 +534,73 @@ std::vector<SimHitIdpr>  TrackerHitAssociator::associateProjectedRecHit(const Pr
   return matched_mono;
 }
 
+void TrackerHitAssociator::associatePhase2TrackerRecHit(const Phase2TrackerRecHit1D* rechit,
+							std::vector<SimHitIdpr> & simtrackid,
+							std::vector<simhitAddr>* simhitCFPos) const
+{
+  //
+  // Phase 2 outer tracker associator
+  //
+  DetId detid=  rechit->geographicalId();
+  uint32_t detID = detid.rawId();
+
+  edm::DetSetVector<PixelDigiSimLink>::const_iterator isearch =  ph2trackerdigisimlink->find(detID); 
+  if(isearch !=  ph2trackerdigisimlink->end()) {  //if it is not empty
+    edm::DetSet<PixelDigiSimLink> link_detset = (*isearch);
+    Phase2TrackerRecHit1D::CluRef const& cluster = rechit->cluster();
+    
+    //check the reference is valid
+    
+    if(!(cluster.isNull())){//if the cluster is valid
+      int minRow = (*cluster).firstStrip();
+      int maxRow = (*cluster).firstStrip() + (*cluster).size();
+      int Col = (*cluster).column();
+      // std::cout << "    Cluster minRow " << minRow << " maxRow " << maxRow << " column " << Col << std::endl;
+      edm::DetSet<PixelDigiSimLink>::const_iterator linkiter = link_detset.data.begin(), linkEnd = link_detset.data.end();
+      int dsl = 0;
+      std::vector<SimHitIdpr> idcachev;
+      std::vector<simhitAddr> CFposcachev;
+      for( ; linkiter != linkEnd; ++linkiter) {
```

Comment:
This could be a range-based for loop over `link_detset.data` (in which case the various `const_iterator` variables defined above can be removed)